test/org/sagebionetworks/bridge/models/schedules/CriteriaScheduleStrategyTest.java

```diff
@@ -0,0 +1,348 @@
+        
+        // context does not have *all* the required groups, last one returned
+        schedule = getScheduleFromStrategy(Sets.newHashSet("group1"));
+        assertEquals(strategyNoCriteria, schedule);
+    }
+    
+    @Test
+    public void filtersOnProhibitedDataGroup() {
+        setUpStrategyWithOneProhibitedDataGroup();
+        setUpStrategyNoCriteria();
+        
+        // Group not prohibited so first schedule returned
+        Schedule schedule = getScheduleFromStrategy(Sets.newHashSet("groupNotProhibited"));
+        assertEquals(strategyWithOneProhibitedDataGroup, schedule);
+        
+        // this group is prohibited so second schedule is returned
+        schedule = getScheduleFromStrategy(Sets.newHashSet("group1"));
+        assertEquals(strategyNoCriteria, schedule);
+    }
+    
+    @Test
+    public void filtersOnProhibitedDataGroups() {
+        setUpStrategyWithProhibitedDataGroups();
+        setUpStrategyNoCriteria();
+        
+        // context has a prohibited group, so the last schedule is returned
+        Schedule schedule = getScheduleFromStrategy(Sets.newHashSet("group1"));
+        assertEquals(strategyNoCriteria, schedule);
+        
+        // context has one of the prohibited groups, same thing
+        schedule = getScheduleFromStrategy(Sets.newHashSet("foo","group1"));
+        assertEquals(strategyNoCriteria, schedule);
+        
+        // context has no prohibited groups, first schedule is returned
+        schedule = getScheduleFromStrategy(Sets.newHashSet());
+        assertEquals(strategyWithProhibitedDataGroups, schedule);
+    }
+    
+    @Test
+    public void noMatchingFilterReturnsNull() {
+        setUpStrategyWithAppVersions();
+        setUpStrategyWithProhibitedDataGroups();
+        
+        User user = getUser();
+        user.setDataGroups(Sets.newHashSet("group1"));
+        ScheduleContext context = new ScheduleContext.Builder()
+                .withClientInfo(ClientInfo.fromUserAgentCache("app/44"))
+                .withUser(user).build();
+
+        Schedule schedule = strategy.getScheduleForUser(plan, context);
+        assertNull(schedule);
+    }
+    
+    @Test
+    public void canMixMultipleCriteria() {
+        setUpStrategyWithAppVersions();
+        setUpStrategyWithOneRequiredDataGroup();
+        setUpStrategyWithProhibitedDataGroups();
+        
+        User user = getUser();
+        user.setDataGroups(Sets.newHashSet("group3"));
+        ScheduleContext context = new ScheduleContext.Builder()
+                .withClientInfo(ClientInfo.fromUserAgentCache("app/44"))
+                .withUser(user).build();
+        
+        // First two don't match because app version is wrong and missing required groups
+        // The last is returned because the context has no prohibited groups
+        Schedule schedule = strategy.getScheduleForUser(plan, context);
+        assertEquals(strategyWithProhibitedDataGroups, schedule);
+    }
```

Comment:
Added, tests both that different criteria are matched, and that the first valid schedule in the list is returned.